### ISSUE TITLE
Restored window size set to maximized after exiting maximized

### ISSUE THREAD

**Issue by J-qak** [atom/atom#10720]
When I exit Atom and window state is restored, its size is remembered; so when I start Atom again, the window starts restored and remembers its size. However, when I exit Atom maximized, the restored window size is set to maximized size; so when I start Atom again, it start maximized and when I click restore, the window size remains just the same (maximized size).

Simply said, nonmaximized window is too often set to maximized size and is just too big to normaly drag around and I have to manually resize window to smaller size each time.

**Reproduction Steps:**

1. Exit Atom maximized
2. Start Atom again and click "restore window"
3. The window size is same as maximized

**Expected behavior:**

Restored window size should be remembered correctly, so you can actually restore window to its last size, not restore it to maximized size.

**Observed behavior:**

I am unable to test this on different OS than Windows 10. I think it's not Electron bug, since VS Code behaves correctly. The reason why this bothers me is that I can't just grab maximized window and drag it, because it's size is same as maximized (= just too big for dragging around screen).

**Screenshots and GIFs**

![output_qijguj](https://cloud.githubusercontent.com/assets/6060970/12868449/857bc078-cd07-11e5-9b7b-768d7669e253.gif)

**Atom version:** 1.4.3
**OS and version:** Windows 10

**Installed packages:**

minimap

**Additional information:**

* Problem can be reproduced in safe mode: Yes
* Problem started happening recently, didn't happen in an older version of Atom: No
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: Yes

**Comment by amytruong** [comment]
I've definitely ran into this issue before on Windows but can't remember if there's an issue for it. :thought_balloon:

**Comment by mastazi** [comment]
Still present as of Atom 1.7.0, despite the Electron update (I was secretly hoping that the update would fix it); using Windows 10 Pro 64 bit.

**Comment by mastazi** [comment]
As I said in #9172:

> I think Atom should follow Visual Studio Code's example (VS Code is also an Electron app) and set a default window size and position which is used in those cases the previous size is not remembered across sessions. At least, with the VS Code behaviour, the "restore down" button is not made useless.

**Comment by Ben3eeE** [comment]
Possible duplicates:
https://github.com/atom/atom/issues/6912 https://github.com/atom/atom/issues/10181

**Comment by Ben3eeE** [comment]
//cc @50Wliu check possible duplicates and maybe incorrect windows label. One of possible duplicates is linux mint.

**Comment by 50Wliu** [comment]
Neither of those are duplicates of this one as those are saying that the window is not maximized, but this one is saying that restoring a maximized window results in a window that has the same size as a maximized one.

**Comment by prasannavl** [comment]
@50Wliu, This is indeed a duplicate of https://github.com/atom/atom/issues/6912. 

`When maximized and then exited, it saves the sizes and not the maximization. And it ends up opening un-maximized, but with the maximized sizes.`

Please close one. The older issue (mine - #6912) also notes the `aero-snap` behavior for the same, while this issue seems to have a little more engagement.

**Comment by Ben3eeE** [comment]
My issue that made me confused is that I exit atom while in full screen (f11) and when restarting it is in fullscreen but then when leaving fullscreen it has the incorrect size and because I have the taskbar in the top I can't access the menu or anything in this state and have to use special steps to resize the window.
Maybe my separate but related?

**Comment by 50Wliu** [comment]
@prasannavl However, @J-qak is saying that the window *is* maximized on restarting Atom:

> when I exit Atom maximized, the restored window size is set to maximized size; so when I start Atom again, it start maximized

This issue is that the *restored* window size isn't set correctly.

**Comment by prasannavl** [comment]
@50Wliu, Makes sense depending on how you look at it I guess. 

The cause of the issue and the expected behavior in #6912 there solves both of these though. Anyway, as you see it fit :)

**Comment by Ben3eeE** [comment]
Is the issue I described also covered by this?

**Comment by 50Wliu** [comment]
I don't think fullscreen state should be remembered - does any other program do this?

**Comment by mastazi** [comment]
@50Wliu my experience on Windows is that most programs remember both of the following:

1. maximised state (wheter the window was maximised upon closing)
2. restored size (the size that the window had before it was maximised)

In Atom, point n. 1 works as expected but point n. 2 doesn't

**Comment by Ben3eeE** [comment]
Actually I can't think of any other programs that restore fullscreen state at the moment.

**Comment by 50Wliu** [comment]
Yeah, I'm bothered constantly by this as well and may take a look at it.  The only other program that I can think of that doesn't handle resizes correctly is Steam.

**Comment by mastazi** [comment]
@Ben3eeE I have just conducted some tests on my Windows 10 Pro machine and some of the programs that remember both maximised state and restored size (restored size is remembered even if the window was maximised upon closing, except where noted using an asterisk) are:

- Chrome 49
- Firefox 43
- MS Edge
- Internet Explorer 11
- Word 2016
- Outlook 2016
- Excel 2016
- PowerPoint 2016
- OneNote 2016
- Photoshop CC 2015
- Illustrator CC 2015 (*)
- VS Code 1.0
- Visual Studio 2015
- Sublime Text 3
- IntelliJ Idea 15

(*) similarly to Atom, this program does not seem to remember the restored size if it was maximised upon closing.

**Comment by Ben3eeE** [comment]
@mastazi My comment was regarding fullscreen state entered by pressing F11 in atom and all modern web browsers on windows not maximized state.

@50Wliu I tested chrome, firefox, opera and internet explorer and it seems split about remembering fullscreen state. Internet explorer and opera does while firefox and chrome does not. Maybe remembering it is unexpected for a text editor I don't know probably a matter of preference.

The window should however always be restored properly when exiting fullscreen and after restarting in fullscreen it does not. The size when exiting fullscreen is bigger than the screen. So this seems like the same issue from a different state only.

**Comment by Ben3eeE** [comment]
@50Wliu I just found https://github.com/atom/atom/issues/8443 regarding remembering fullscreen state.

**Comment by skylize** [comment]
I am so sad that this issue is still open.  😢

> my experience on Windows is that most programs remember both of the following:
> 
> maximised state (wheter the window was maximised upon closing)
> restored size (the size that the window had before it was maximised)
>
> In Atom, point n. 1 works as expected but point n. 2 doesn't

Some apps remember full-screen and some don't. That is less important to expected behavior. Remembering the last restored size is much more important for living up to user expectations. 

Should we remember or forget that the window was maxed when I closed it? I don't much care. 

But I get super irritated 😖 when I must _yet again_ drag an Atom window away from the side of the screen ... just so I can shrink it down to at least be smaller than the screen ... just so I can drag it to more or less where I want it to be on the screen ... so I can finally start working on carefully and painstakingly resizing it to the same size as I had it before I closed it, just because I forgot to un-maximize it before close.

**Comment by ks00x** [comment]
This anoying bug is still alive in V1.25.1 on both windows 7 and 10

**Comment by Coalpha** [comment]
This bug is also still present in V1.28.1 on Windows 10. This happens to me whenever I don't resize the window after going into a new project folder.

**Comment by natonboy** [comment]
Sooo....hello 1.30.0 x64 (Windows 10)

**Comment by eithan1231** [comment]
So goddamn annoying. Atom is the best IDE I have used, but this bug is beginning to become a deal breaker...

**Comment by KhalilOuali** [comment]
July 2020 - Bug still not fixed in Atom 1.49.0 x64 (latest version at time of comment) for Windows 10. 
Too bad, Atom is my favorite editor but this bug can be frustrating at times.
Here's hoping it's fixed soon.

**Comment by LupinIII** [comment]
Still broken and annoying in 1.54.0 x64.
